src/svg.rs

```diff
@@ -33,13 +35,39 @@ impl SvgVersion {
 declare_surface!(SvgSurface, SurfaceType::Svg);
 
 impl SvgSurface {
-    pub fn new<P: AsRef<Path>>(width: f64, height: f64, path: P) -> Result<SvgSurface, Status> {
-        let path = path.as_ref().to_string_lossy().into_owned();
-        let path = CString::new(path).unwrap();
+    pub fn new<P: AsRef<Path>>(
+        width: f64,
+        height: f64,
+        path: Option<P>,
+    ) -> Result<SvgSurface, Status> {
+        #[cfg(not(windows))]
+        let path = {
+            let path = path.map(|p| {
+                CString::new(p.as_ref().as_os_str().as_bytes())
+                    .expect("Invalid path with NULL bytes")
+            });
+            path.map(|p| p.as_ptr()).unwrap_or(ptr::null())
```

Comment:
The path will be freed once this block is done, and then you store a dangling pointer in path in the outer scope. Just return the `Option<CString>` from this block, and do the `as_ptr()` part for both Windows/UNIX just before calling the C function.